grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -244,6 +245,49 @@ class StockMovementController {
         redirect(action: "list", params: params << ["flash": flash as JSON])
     }
 
+    def updateStatus = {
+        StockMovement stockMovement = stockMovementService.getStockMovement(params.id)
+
+        try {
```

Comment:
I really don't like throwing that try/catch everywhere - lately we've been doing that very much. Just wanted to mention that a potential rollback might not take place in Grails 1 when we catch exceptions.
Is it in requirements that after an exception we want to show the same view page?

Reply:
It was not mentioned if it should be the same page so it was my own interpretation.

It was mentioned that reject and approve buttons should be disabled if a request is already approved/rejected.
This try catch was more like a failsafe I added so I think I can remove it and let the generic handler do it's job and render generic error page.